src/widgets/Header.tsx

```diff
@@ -0,0 +1,97 @@
+import WidgetBase from '@dojo/framework/widget-core/WidgetBase';
+import Resize, { ContentRect } from '@dojo/framework/widget-core/meta/Resize';
+import { tsx } from '@dojo/framework/widget-core/tsx';
+import Link from '@dojo/framework/routing/ActiveLink';
+
+const logo = require('../assets/logo.svg');
+
+import * as css from './Header.m.css';
+
+const pages = ['Blog', 'Documentation', 'Examples', 'Playground', 'Community'];
+
+export default class Menu extends WidgetBase {
+	protected expanded = false;
+
+	protected get isSmall(): boolean {
+		const { isSmall } = this.meta(Resize).get('root', {
+			isSmall: this.smallPredicate
+		});
+
+		if (!isSmall) {
+			this.expanded = false;
+		}
+
+		return isSmall;
+	}
+
+	protected smallPredicate(contentRect: ContentRect): boolean {
+		return contentRect.width < 768;
+	}
+
+	protected toggle() {
+		this.expanded = !this.expanded;
+		this.invalidate();
+	}
+
+	protected close() {
+		this.expanded = false;
+		this.invalidate();
+	}
+
+	protected onKeyDown(event: KeyboardEvent) {
+		if (event.key === 'Escape') {
+			this.close();
+		}
+	}
+
+	protected renderToggleButton(shouldRender: boolean) {
+		if (shouldRender) {
+			return (
+				<button
+					key="toggleButton"
+					onclick={this.toggle}
+					classes={css.toggleButton}
+					aria-expanded={this.expanded}
+				>
+					<span classes={css.srOnly}>Menu</span>
+					<div classes={css.toggleBar} />
+				</button>
+			);
+		}
+	}
+
+	protected render() {
+		const { isSmall, expanded } = this;
+		const rootClasses = [css.root];
+		isSmall && rootClasses.push(css.responsive);
+		expanded && rootClasses.push(css.expanded);
+		return (
+			<header key="root" classes={rootClasses} onkeydown={this.onKeyDown}>
+				<div classes={[css.left]}>
+					<span classes={css.toggleButtonContainer}>{this.renderToggleButton(isSmall)}</span>
```

Comment:
Might be clearer if we use a ternary here rather than logic in the `renderToggleButton`. 

```ts
isSmall ? this.renderToggleButton : null
```